enterprise-suite/console_test/prometheus_test.go

```diff
@@ -0,0 +1,264 @@
+			Expect(prom.HasData(fmt.Sprintf("health{name=\"%v\"}", metric))).To(BeTrue())
+		}
+
+		// Coherency
+
+		// Data with "es_workload" should also have a "namespace" label
+		Expect(prom.HasData("count({es_workload=~\".+\", namespace=\"\", name!~\"node.*|kube_node.*\", __name__!~\"node.*|kube_node.*\"})")).To(BeFalse())
+		// Health should have "es_workload" label, with a few known exceptions
+		Expect(prom.HasData("health{es_workload=\"\", name!~\"node.*|kube_node.*|prometheus_target_down|scrape_time\"}")).To(BeFalse())
+		// kube_pod_info must have es_workload labels
+		Expect(prom.HasData("kube_pod_info{es_workload=~\".+\"}")).To(BeTrue())
+		Expect(prom.HasData("kube_pod_info{es_workload=\"\"}")).To(BeFalse())
+		// kube data mapped pod to workload labels
+		Expect(prom.HasData("{__name__=~ \"kube_.+\", pod!=\"\", es_workload=\"\"}")).To(BeFalse())
+		// All container data have a workload label
+		Expect(prom.HasData("{__name__=~\"container_.+\", es_workload=\"\"}")).To(BeFalse())
+		// All targets should be reachable
+		Expect(prom.HasData("up{kubernetes_name != \"es-test-service-with-only-endpoints\"} == 1")).To(BeTrue())
+		Expect(prom.HasData("up{kubernetes_name != \"es-test-service-with-only-endpoints\"} == 0")).To(BeFalse())
+
+		// kube-state-metrics
+
+		kubeStateMetrics := []string{
+			"kube_pod_info",
+			"kube_pod_ready",
+			"kube_pod_container_restarts_rate",
+			"kube_pod_failed",
+			"kube_pod_not_running",
+			"kube_workload_generation_lag",
+		}
+
+		for _, metric := range kubeStateMetrics {
+			Expect(prom.HasData(metric)).To(BeTrue())
+		}
+
+		kubeStateHealth := []string{
+			"kube_container_restarts",
+			"kube_pod_not_ready",
+			"kube_pod_not_running",
+			"kube_workload_generation_lag",
+		}
+
+		for _, metric := range kubeStateHealth {
+			Expect(prom.HasData(fmt.Sprintf("model{name=\"%v\"}", metric))).To(BeTrue())
+			Expect(prom.HasData(fmt.Sprintf("health{name=\"%v\"}", metric))).To(BeTrue())
+		}
+
+		// TODO: Check kube-state-metrics logs
+
+		// node-exporter
+
+		nodeMetrics := []string{
+			"node_cpu",
+			"node_memory_MemAvailable",
+			"node_filesystem_free",
+			"node_network_transmit_errs",
+		}
+
+		for _, metric := range nodeMetrics {
+			Expect(prom.HasData(metric)).To(BeTrue())
+		}
+
+		// TODO: Check node-exporter logs
+
+		// App tests
+
+		// App via 'Pod' service discovery
+		appInstancesQuery := fmt.Sprintf("count( count by (instance) (ohai{es_workload=\"es-test\", namespace=\"%v\"}) ) == 2", promTestNamespace)
+		err := util.WaitUntilTrue(func() bool {
+			return prom.HasData(appInstancesQuery)
+		})
+		Expect(err).To(Succeed())
+
+		// Pod discovery - automatic metrics should gain an es_monitor_type label when a custom monitor is created
+		Expect(esMonitor.Make("es-test/my_custom_monitor", "up")).To(Succeed())
+		err = util.WaitUntilTrue(func() bool {
+			return prom.HasModel("my_custom_monitor")
+		})
+		Expect(err).To(Succeed())
```

Comment:
Is there a way to make nicer error messages when these tests fail? It would be nice to see the query for example, and what it returned (or didn't return). Maybe we need another utility method which returns a more detailed error. I think it will be quite difficult to figure out why these failed on a Travis build without that detail.

Reply:
Improved it a bit with additional message param in `WaitUntilTrue`